From ef2bf10dd3826636443d68100dce88da2030a0e2 Mon Sep 17 00:00:00 2001
From: Vitaly Margolin <vmargolin@habana.ai>
Date: Thu, 22 May 2025 13:30:10 +0300
Subject: [PATCH] [SW-229877] g3: serdes: move serdes partition in flash

Move Serdes partition in flash from address
1C0000 to 1D0000. We need to move it because a problem
with block 1C0000 corruption on secured setups, when
EFUSE setting is performed. After this problem
will be resolved, it will be moved back

Change-Id: I7c9f26b402f5cbdb33d730cac38d52b897ddb885
Signed-off-by: Vitaly Margolin <vmargolin@habana.ai>
---

diff --git a/board/habanalabs/gaudi3/create-serdes-config.sh b/board/habanalabs/gaudi3/create-serdes-config.sh
index cb82785..15267cd 100755
--- a/board/habanalabs/gaudi3/create-serdes-config.sh
+++ b/board/habanalabs/gaudi3/create-serdes-config.sh
@@ -10,7 +10,7 @@
 IMAGE_BIN=${1}
 
 SPECS_FILE=${SPECS_EMB_PROJ_DIR}/${DEV_PREFIX}/${DEV_PREFIX}_mem_addr_dt.h
-SERDES_OFFSET="SPI_SERDES_CONFIG_OFFSET"
+SERDES_OFFSET="SPI_SERDES_CONFIG_OFFSET_NEW"
 FLASH_OFFSET=0x"$(grep -E "^\#define[[:space:]]+$SERDES_OFFSET[[:space:]]+" "$SPECS_FILE" | sed "s/^\#define[[:space:]]\+$SERDES_OFFSET[[:space:]]\+//")"
 
 VERSION=`cat ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/agent-fwl-custom/build/buildversion-habanalabs`
diff --git a/board/habanalabs/gaudi3_arc/agent-fw_loader.its b/board/habanalabs/gaudi3_arc/agent-fw_loader.its
index c9e37bc..66c0950 100644
--- a/board/habanalabs/gaudi3_arc/agent-fw_loader.its
+++ b/board/habanalabs/gaudi3_arc/agent-fw_loader.its
@@ -46,7 +46,7 @@
 			data = /incbin/("gaudi3-serdes-config.bin");
 			type = "standalone";
 			compression = "none";
-			offset = <0x001C0000>;
+			offset = <0x001D0000>;
 			hash {
 				algo = "sha384";
 			};
diff --git a/board/habanalabs/gaudi3_arc/post-image-spi.sh b/board/habanalabs/gaudi3_arc/post-image-spi.sh
index 5b2f15b..6347012 100755
--- a/board/habanalabs/gaudi3_arc/post-image-spi.sh
+++ b/board/habanalabs/gaudi3_arc/post-image-spi.sh
@@ -440,7 +440,7 @@
 
 function  print_serdes_image_desc() {
 	local img_num=10
-	local serdes_partition_sector_num=28
+	local serdes_partition_sector_num=29
 	local flash_sector_size=0x10000
 	local s_off=$(($serdes_partition_sector_num * $flash_sector_size))
 
